cmd/snap/cmd_run.go

```
@@ -1206,5 +1221,20 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 	}
 }
 
+func getSnapDirOptions() *dirs.SnapDirOptions {
+	var opts dirs.SnapDirOptions
+
+	_, err := os.Stat(features.HiddenSnapFolder.ControlFile())
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			logger.Noticef("WARNING: cannot get conf %q, assuming it's unset: %s", features.HiddenSnapFolder, err)
```

we need to return an error